mfkdf2/src/setup/factors/mod.rs

```diff
@@ -1,87 +1,88 @@
 use serde::{Deserialize, Serialize};
 use serde_json::Value;
-// pub mod hmacsha1;
+pub mod hmacsha1;
 pub mod hotp;
+pub mod ooba;
+pub mod passkey;
 pub mod password;
-// pub mod question;
-// pub mod stack;
-// pub mod totp;
-// pub mod uuid;
+pub mod question;
+pub mod stack;
+pub mod totp;
+pub mod uuid;
 
-// pub use hmacsha1::hmacsha1;
+pub use hmacsha1::hmacsha1;
 pub use hotp::hotp;
+pub use passkey::passkey;
 pub use password::password;
-// pub use question::question;
-// pub use stack::stack;
-// pub use uuid::uuid;
+pub use question::question;
+pub use stack::stack;
+pub use totp::totp;
+pub use uuid::uuid;
 
 #[derive(Clone, Debug, Serialize, Deserialize, uniffi::Enum)]
-pub enum FactorType {
+pub enum FactorSetupType {
   Password(password::Password),
   HOTP(hotp::HOTP),
-  // Question(question::Question),
-  // UUID(uuid::UUID),
-  // HmacSha1(hmacsha1::HmacSha1),
-  // TOTP(totp::TOTP),
-  // Stack(stack::Stack),
+  Question(question::Question),
+  UUID(uuid::UUID),
+  HmacSha1(hmacsha1::HmacSha1),
+  TOTP(totp::TOTP),
+  OOBA(ooba::Ooba),
+  Passkey(passkey::Passkey),
+  Stack(stack::Stack),
 }
 
-impl FactorType {
-  pub fn inner(&self) -> &dyn FactorTrait {
+impl FactorSetupType {
```

Comment:
I don't know if we need two types for this enum ultimately. You may really just want to have this be an enum for factors and only implement `fn inner()` and `fn inner_mut()`. 

This was why I had one `Factor` trait to begin with.